passflip/input.py

```diff
@@ -1,23 +1,21 @@
 from getpass import getpass
 
 
-def create_prompt_message(message, again=False):
-    prompt_message = message
-    if again:
-        prompt_message += " again"
-    prompt_message += ": "
-    return prompt_message
+def prompt(message, validate):
+    first_input = getpass(message + ': ')
+    if len(first_input) == 0:
+        return None
+    if not validate:
+        return first_input
+    second_input = getpass(message + 'again: ')
+    if first_input != second_input:
+        first_input = None
+    return first_input
```

Comment:
What does it return? Sometimes it returns the first thing typed in, sometimes it returns `None`. I think the `first_input = None` rather than `return None` is confusing me more.